fix(icons): fix icons invisible in high contrast mode

wkeese opened the pull request:
Closes #17147, #18830, #19015, #19023, and partially fixes #19140.
Also fixes problems like showing the selected tab in UIShell / SidePanel.
Likely also fixes similar problems in @carbon/ibm-products.
Refs #8489.

If the system has a dark high-contrast theme, then icons need to be displayed in white (or a light color), and vice-versa.  Otherwise, icons become invisible (black on black or white on white). Importantly, the icon color needs to be set relative to the system's high contrast theme _regardless of which Carbon theme is selected_.

Although Carbon already has a mixin for this, it is not used consistently, and it would be labor-intensive, bloat the code, and error-prone to do so.   Because you basically would need to call the mixin every time you set a `fill`, which is done hundreds of times.   For example:

```scss
.#{$prefix}--accordion__arrow,
.#{$prefix}--accordion__item--active .#{$prefix}--accordion__arrow {
  fill: $icon-primary;
  @include high-contrast-mode('icon-fill');
}
```

And as I wrote in #19015, you also need to worry about CSS precedence, so you might think you've fixed a bug, but it isn't fixed in all cases.

So instead, this PR goes to the root of the issue by just overriding the value of `--cds-icon-primary` CSS property referenced by the `$icon-primary` SCSS  variable.   Likewise for similar CSS properties.   For example, consider [this story](https://react.carbondesignsystem.com/?path=/story/components-iconbutton--default&args=kind:ghost&globals=theme:g100), in the G100 theme.  There's this existing CSS:

```css
[data-carbon-theme=g100] {
  --cds-icon-primary: #f4f4f4;
  --cds-icon-secondary: ...
  ...
}
```

But in high contrast mode, it's now overridden by:

```css
@media screen and (-ms-high-contrast: active), (forced-colors: active) {
    [data-carbon-theme=g100] {
        --cds-icon-primary: ButtonText;
        --cds-icon-secondary: ButtonText;
        --cds-icon-disabled: GrayText;
        --cds-icon-on-color-disabled: GrayText;
        --cds-icon-inverse: SelectedItemText;
        --cds-icon-on-color: SelectedItemText;
        ...
    }
}
```

I also removed the current 27 calls to:

```scss
@include high-contrast-mode('icon-fill');
```

Plus a few other places that were manually setting `fill` to `ButtonText`, but now no longer need to do so.

### Changelog

**New**

- Added CSS to override the values of icon CSS properties (ex: `--cds-icon-primary`) when the system is in high contrast mode.

#### Testing / Reviewing

Tested in Storybook on Windows 11, using the Aquatic and Desert themes.

Note that you can emulate forced-mode on Mac in Chrome (see https://cssence.com/2024/forced-colors-mode-strategies/), which is useful if you don't have a PC handy.   Just open Chrome DevTools, hit Ctrl/Cmd+Shift+P, type “forced”, and you’ll find the option to "Emulate CSS forced-colors: active".

## PR Checklist

As the author of this PR, before marking ready for review, confirm you:

- [x] Reviewed every line of the diff
- [x] Updated documentation and storybook examples
- [N/A] Wrote passing tests that cover this change
- [x] Addressed any impact on accessibility (a11y)
- [x] Tested for cross-browser consistency
- [x] Validated that this code is ready for review and status checks should pass

More details can be found in the [pull request guide](https://github.com/carbon-design-system/carbon/blob/main/docs/guides/reviewing-pull-requests.md)

wkeese committed:
fix(icons): fix icons invisible in high contrast mode

If the system has a dark high-contrast theme, then icons need to be displayed
in white (or a light color), and vice-versa.  Otherwise, icons become invisible.
Importantly, this must happen regardless of the Carbon theme setting.

Fixes #17147, #18830, #19015, #19023, #19140.

wkeese committed:
fix(icons): fix tooltip story

Use proper color for icon, not background color.

heloiselui committed:
fix: remove unexpected shadow from web-component button (#19347)

sojinantony01 committed:
fix: controlled tag component for selected prop #19228 (#19231)

* fix: controlled tag component for selected prop #19228

* fix: fix tests for tag

* fix: controlled tag component for selected prop #19228

* fix: controlled tag added a default selected prop and tests #19228

* fix: controlled tag added a default selected prop and tests #19228

* fix: controlled tag added a default selected prop and tests #19228

* fix: controlled tag fix tests #19228

* fix: add profile

* fix: add profile

heloiselui reviewed (commented):
Great, thank you so much for working on this!

I've been reviewing all the issues and your updates:

For `@include high-contrast-mode('icon-fill') `- try removing it in this PR if possible, so I can review it here.

I tested #17147, #18830, #19015, and #19023 - they all look good, thanks 🎉

I tested #19140 and it still shows the same behavior as the issue. I recorded a quick video so you can take a look. I also saw that @mariyageorge01 made some changes in her PR, I tested it and it works as expected. Maybe we can wait for your PR to be merged first, and then she can continue with her updates? What do you think?

https://github.com/user-attachments/assets/b6858bcf-0454-43e9-850e-ca3096aeedfa

warrenmblood committed:
fix(date-picker): clear flatpickr instance on reset (#19239)

* fix(date-picker): clear flatpickr instance on reset

* fix(date-picker): clear flatpickr instance on reset

* fix(date-picker): update comments and storybook docs to clarify changes

* fix(date-picker): update comments and storybook docs to clarify changes

* Update packages/react/src/components/DatePicker/DatePicker.tsx

Co-authored-by: kennylam <909118+kennylam@users.noreply.github.com>

---------

Co-authored-by: kennylam <909118+kennylam@users.noreply.github.com>

wkeese committed:
fix(icons): remove vestigial high-contrast-mode('icon-fill')

When you use standard tokens like $icon-primary you don't
need it anymore.

Note: I didn't convert $support-error and $support-warning usages
to a system color. Previously the inline notification icon 🚫 was converted
from red to ButtonText.  Also $support-error and $support-warning usages in
_list-box.scss.

Also, inline-notification has a dodgy use of $black-100 that I didn't change.

wkeese committed:
fix(icons): handle weird values for fill

Replace $text-primary with $icon-primary as they are the same.

Likewise $tab-text-disabled --> $icon-disabled.

Updated _theme.scss to handle the other weird fill settings.

wkeese committed:
fix(icons): fix comment

wkeese committed:
fix(icons): fix background colors

preetibansalui committed:
fix: modal footer buttons were not fully visible in zoom to 400 (#19288)

* fix: modal footer buttons were not fully visible in zoom to 400

* fix: added test story

* fix: remove test story

---------

Co-authored-by: Heloise Lui <71858203+heloiselui@users.noreply.github.com>

wkeese committed:
fix(icons): fix checkmark

wkeese committed:
fix(icons): remove unused imports of high-contrast-mode

wkeese committed:
fix(icons): another inverted background

wkeese committed:
Merge branch 'main' into high-contrast-icons

wkeese committed:
chore(icons): fix merge error

heloiselui reviewed (approved):
I tested this again and everything works as expected. Thanks again @wkeese! 🚀

annawen1 reviewed (approved):
tested and LGTM as well!
